flowmachine/flowmachine/features/spatial/location_cluster.py

```diff
@@ -187,6 +188,13 @@ def __init__(
 
         super().__init__()
 
+    @property
+    def column_names(self) -> List[str]:
+        if self.aggregate:
```

Comment:
Not relevant to this PR, rather a note to self: the name `self.aggregate` is confusing because it doesn't indicate whether it's a boolean or some sort of object representing an aggregate. Would be useful to rename this for clarity (or rather, we probably shouldn't be doing the aggregation inside this class in the first place.)

Reply:
This class needs a refactor, and probably splitting into at least three I think.